overlord/servicestate/quota_control.go

```diff
@@ -58,7 +68,59 @@ func EnsureQuotaUsability() (restore func()) {
 	}
 }
 
+func setMemoryCGroupStatus() {
+	memoryCGroupError = checkMemoryCGroup()
+}
+
+// added to enable path update for testing purposes
+func SetCGroupsFilePath(path string) {
+	CGroupsFilePath = path
+	setMemoryCGroupStatus()
+}
+
+// since the control groups can be enabled/disabled without the kernel config the only
+// way to identify the status of memory control groups is via /proc/cgroups
+// "cat /proc/cgroups | grep memory" returns the active status of memory control group
+// and the 3rd parameter is the status
+// 0 => false => disabled
+// 1 => true => enabled
+func checkMemoryCGroup() error {
+	cgroupsFile, err := os.Open(CGroupsFilePath)
+	if err != nil {
+		return fmt.Errorf("cannot open cgroups file: %v", err)
+	}
+	defer cgroupsFile.Close()
+	scanner := bufio.NewScanner(cgroupsFile)
+	scanner.Split(bufio.ScanLines)
```

Comment:
(nitpick) this is not needed: `go doc bufio.Scanner.Split` says it's the default.